pkg/auth/providers/oidc/oidc_provider.go

```diff
@@ -0,0 +1,385 @@
+	}
+	if principalID == UserType {
+		p = v3.Principal{
+			ObjectMeta:    metav1.ObjectMeta{Name: principalType + "://" + externalID},
+			DisplayName:   externalID,
+			LoginName:     externalID,
+			PrincipalType: UserType,
+			Provider:      o.ProviderName,
+		}
+	} else {
+		p = o.GroupToPrincipal(externalID)
+	}
+	p = o.ToPrincipalFromToken(principalType, p, &token)
+	return p, nil
+}
+
+func (o *Provider) TransformToAuthProvider(authConfig map[string]interface{}) (map[string]interface{}, error) {
+	p := common.TransformToAuthProvider(authConfig)
+	p[publicclient.OIDCProviderFieldRedirectURL] = o.GetRedirectURL(authConfig)
+	return p, nil
+}
+
+func (o *Provider) GetRedirectURL(config map[string]interface{}) string {
+	return fmt.Sprintf(
+		"%s?client_id=%s&response_type=code&redirect_uri=%s",
+		config["authEndpoint"],
+		config["clientId"],
+		config["rancherUrl"],
+	)
+}
+
+func (o *Provider) RefetchGroupPrincipals(principalID string, secret string) ([]v3.Principal, error) {
+	return nil, errors.New("[generic oidc]: not implemented")
+}
+
+func (o *Provider) CanAccessWithGroupProviders(userPrincipalID string, groupPrincipals []v3.Principal) (bool, error) {
+	config, err := o.GetOIDCConfig()
+	if err != nil {
+		logrus.Errorf("[generic oidc]: error fetching OIDCConfig: %v", err)
+		return false, err
+	}
+	allowed, err := o.UserMGR.CheckAccess(config.AccessMode, config.AllowedPrincipalIDs, userPrincipalID, groupPrincipals)
+	if err != nil {
+		return false, err
+	}
+	return allowed, nil
+}
+
+func (o *Provider) UserToPrincipal(userInfo *oidc.UserInfo, info ClaimInfo) v3.Principal {
+	displayName := info.Name
+	if displayName == "" {
+		displayName = userInfo.Email
+	}
+	p := v3.Principal{
+		ObjectMeta:    metav1.ObjectMeta{Name: o.ProviderName + "_" + UserType + "://" + userInfo.Subject},
+		DisplayName:   displayName,
+		LoginName:     userInfo.Email,
+		Provider:      o.ProviderName,
+		PrincipalType: UserType,
+		Me:            false,
+	}
+	return p
+}
+
+func (o *Provider) GroupToPrincipal(groupName string) v3.Principal {
+	p := v3.Principal{
+		ObjectMeta:    metav1.ObjectMeta{Name: o.ProviderName + "_" + GroupType + "://" + groupName},
+		DisplayName:   groupName,
+		Provider:      o.ProviderName,
+		PrincipalType: GroupType,
+		Me:            false,
+	}
+	return p
+}
+
+func (o *Provider) ToPrincipalFromToken(principalType string, princ v3.Principal, token *v3.Token) v3.Principal {
+	if principalType == UserType {
+		princ.PrincipalType = UserType
+		if token != nil {
+			princ.Me = o.IsThisUserMe(token.UserPrincipal, princ)
+			if princ.Me {
+				princ.LoginName = token.UserPrincipal.LoginName
+				princ.DisplayName = token.UserPrincipal.DisplayName
+			}
+		}
+	} else {
+		princ.PrincipalType = GroupType
+		if token != nil {
+			princ.MemberOf = o.TokenMGR.IsMemberOf(*token, princ)
+		}
+	}
+	return princ
+}
+
+func (o *Provider) SaveOIDCConfig(config *v32.OIDCConfig) error {
```

Comment:
This doesn't need to be exported

Reply:
removed the functions from being exported that were not longer used outside the oidc package